src/js_user.hpp

```diff
@@ -303,5 +307,54 @@ void UserClass<T>::refresh_custom_data(ContextType ctx, ObjectType this_object,
         }));
 }
 
+template<typename T>
+void UserClass<T>::push_register(ContextType ctx, ObjectType this_object, Arguments& args, ReturnValue &return_value) {
+    args.validate_count(3);
+    auto user = get_internal<T, UserClass<T>>(ctx, this_object);
+    auto service = Value::validated_to_string(ctx, args[0], "service");
+    auto token = Value::validated_to_string(ctx, args[1], "token");
+    auto callback = Value::validated_to_function(ctx, args[2], "callback");
+
+    user->m_app->push_notification_client(service).register_device(
+        token,
+        *user,
+        realm::util::EventLoopDispatcher([ctx = Protected(Context<T>::get_global_context(ctx)),
+                            callback = Protected(ctx, callback),
+                            this_object = Protected(ctx, this_object)]
+                           (util::Optional<app::AppError> error) {
+            HANDLESCOPE(ctx);
+            Function::callback(ctx, callback, this_object, {
+                !error ? Value::from_undefined(ctx) : Object::create_obj(ctx, {
+                    {"message", Value::from_string(ctx, error->message)},
+                    {"code", Value::from_number(ctx, error->error_code.value())},
+                }),
+            });
+        }));
+}
+
+template<typename T>
+void UserClass<T>::push_deregister(ContextType ctx, ObjectType this_object, Arguments& args, ReturnValue &return_value) {
```

Comment:
What's the difference to to "register" function above? They look identical.

Reply:
😬 Yikes! That would be a bug. It is supposed to be calling `deregister_device()` on like 343.

Reply:
So seems like the tests needs an update as well then ;-)
Isn't it possible to reuse some of that code so it's clear where there is supposed to be a difference :-)

Reply:
The API for `deregisterDevice` changed upsteam in object-store, so I'll add an update to the submodule to this PR so we use the new API https://github.com/realm/realm-object-store/commit/1f3290a840a114cc21df78f2fb5fdc722fe1cd7d#diff-b215f5f5ce1cec59d075f06fbab4510e

> Isn't it possible to reuse some of that code so it's clear where there is supposed to be a difference

Probably. I have some ideas on refactors that can clean these repetitive methods up a bit, but so far I've only done the `promisify()` part on the JS side. I'll probably make a separate PR to clean up the CPP side at some point.